Prisma generate should fail for Typescript users below 4.1

## Problem

<!-- A clear and concise description of what the problem is. Ex. I'm always frustrated when [...] -->

The  Prisma Client's generated types requires Typescript 4.1 and above. Right now we show a warning, but should fail because it's too easy to miss this hard requirement.

```
npm run build      
> website-v8@1.0.0 build /Users/m/dev/src/github.com/prisma/homepage-v8
> npm run generate:explore && next build
> website-v8@1.0.0 generate:explore /Users/m/dev/src/github.com/prisma/homepage-v8
> prisma generate --schema=./ui/explore/schema.prisma && node ./scripts/generate-explore.js
Prisma schema loaded from ui/explore/schema.prisma
**prisma:warn Your typescript version is 4.0.3, which is outdated. Please update it to 4.1.0 or newer in order to use Prisma Client.**
✔ Generated Prisma Client (2.19.0) to ./node_modules/@prisma/client in 152ms
You can now start using Prisma Client in your code. Reference: https://pris.ly/d/client

import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()

Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db
Failed to compile.
./node_modules/.prisma/client/index.d.ts:11:33
Type error: Type expected.
   9 | type UnwrapPromise<P extends any> = P extends Promise<infer R> ? R : P
  10 | type UnwrapTuple<Tuple extends readonly unknown[]> = {
> 11 |   [K in keyof Tuple]: K extends `${number}` ? Tuple[K] extends PrismaPromise<infer X> ? X : UnwrapPromise<Tuple[K]> : UnwrapPromise<Tuple[K]>
     |                                 ^
  12 | };
  13 | 
  14 | 
info  - Creating an optimized production build .npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! website-v8@1.0.0 build: `npm run generate:explore && next build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the website-v8@1.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/m/.npm/_logs/2021-03-19T12_46_24_629Z-debug.log
```

## Suggested solution

<!-- A clear and concise description of what you want to happen. -->

Show an error during `prisma generate` if the user is using a Typescript version that's less than 4.1. 

**Warning** that this is only a problem for Typescript users, not Javascript users. We shouldn't make this a hard requirement for Javascript users.

--- a/src/packages/sdk/src/predefinedGeneratorResolvers.ts
+++ b/src/packages/sdk/src/predefinedGeneratorResolvers.ts
@@ -167,7 +167,7 @@ function checkTypeScriptVersion() {
     if (output.stdout) {
       const currentVersion = output.stdout.split(' ')[1]
       if (semverLt(currentVersion, minVersion)) {
-        logger.error(
+        throw new Error(
           `Your ${chalk.bold(
             'typescript',
           )} version is ${currentVersion}, which is outdated. Please update it to ${chalk.bold(
@@ -175,7 +175,6 @@ function checkTypeScriptVersion() {
           )} or ${chalk.bold('newer')} in order to use Prisma Client.`,
         )
       }
-      process.exit(1)
     }
   } catch (e) {
     // They do not have TS installed, we ignore (example: JS project)

--- a/src/packages/sdk/src/predefinedGeneratorResolvers.ts
+++ b/src/packages/sdk/src/predefinedGeneratorResolvers.ts
@@ -167,7 +167,7 @@ function checkTypeScriptVersion() {
     if (output.stdout) {
       const currentVersion = output.stdout.split(' ')[1]
       if (semverLt(currentVersion, minVersion)) {
-        logger.warn(
+        logger.error(
           `Your ${chalk.bold(
             'typescript',
           )} version is ${currentVersion}, which is outdated. Please update it to ${chalk.bold(
@@ -175,6 +175,7 @@ function checkTypeScriptVersion() {
           )} or ${chalk.bold('newer')} in order to use Prisma Client.`,
         )
       }
+      process.exit(1)
     }
   } catch (e) {
     // They do not have TS installed, we ignore (example: JS project)

